### PR TITLE
chore: Add "Deploy Preview" GitHub workflow

### PR DESCRIPTION
**Pull request by LekoArts**
## Description

This PR adds a new GitHub workflow called `preview`. You'll be able to comment `!preview` on a PR to build `playground/nextjs` with the current state of your PR. This also includes `clerk-js`.

How it works:

1. Build the monorepo packages
2. Copy playground/nextjs to a temp dir and use secco to install monorepo packages (and rest of deps) inside newly copied test site
3. Copy clerk-js/dist to public/clerk-js of test site and set env var to use that CLERK_JS_URL
4. Build site with Vercel
5. Upload site to Vercel
6. Add comment to PR to point towards preview

**Note: Only members of Clerk Vercel Team will be able to access the deploy preview**

Parts of the logic are copied from https://github.com/clerk/clerk-docs/blob/main/.github/workflows/preview.yml

## Checklist

- [x] `npm test` runs as expected.
- [x] `npm run build` runs as expected.
- [ ] (If applicable) [JSDoc comments](https://jsdoc.app/about-getting-started.html) have been added or updated for any package exports
- [ ] (If applicable) [Documentation](https://github.com/clerk/clerk-docs) has been updated

## Type of change

- [ ] 🐛 Bug fix
- [x] 🌟 New feature
- [ ] 🔨 Breaking change
- [ ] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

## Packages affected

- [ ] `@clerk/backend`
- [ ] `@clerk/chrome-extension`
- [ ] `@clerk/clerk-js`
- [ ] `@clerk/clerk-expo`
- [ ] `@clerk/fastify`
- [ ] `gatsby-plugin-clerk`
- [ ] `@clerk/localizations`
- [ ] `@clerk/nextjs`
- [ ] `@clerk/clerk-react`
- [ ] `@clerk/remix`
- [ ] `@clerk/clerk-sdk-node`
- [ ] `@clerk/shared`
- [ ] `@clerk/themes`
- [ ] `@clerk/types`
- [x] `build/tooling/chore`

### CODE REVIEW

**Comment by LekoArts** [comment]
!preview

**Comment by LekoArts** [comment]
!preview

File: .github/workflows/preview.yml [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+    types: [created]
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}-${{ github.actor }}
+  cancel-in-progress: true
+
+jobs:
+  preview:
+    if: ${{ startsWith(github.event.comment.body, '!preview') && github.repository == 'clerk/javascript' && github.event.issue.pull_request }}
+    runs-on: ${{ vars.RUNNER_NORMAL }}
```
Comment (LekoArts) [inline .github/workflows/preview.yml:18]: IMO it's not urgent so no need to spend more money than necessary

File: .github/workflows/preview.yml [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+    timeout-minutes: ${{ fromJSON(vars.TIMEOUT_MINUTES_NORMAL) }}
+
+    env:
+      VERCEL_ORG_ID: ${{ secrets.VERCEL_CLERK_PROD_ORG_ID }}
+      VERCEL_PROJECT_ID: ${{ secrets.VERCEL_JS_PREVIEW_PROJECT_ID }}
```
Comment (LekoArts) [inline .github/workflows/preview.yml:23]: Env vars that the Vercel API uses (kind of undocumented feature at this point)

File: .github/workflows/preview.yml [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+
+      - name: Build release
+        run: npx turbo build $TURBO_ARGS --force
+
+      - name: Get date
+        run: |
+          echo "DATE=$(date -u +"%b %d, %Y %I:%M %p")" >> $GITHUB_ENV
+
+      - name: Install site in isolation
+        run: node scripts/install-site-in-isolation.mjs playground/nextjs
```
Comment (LekoArts) [inline .github/workflows/preview.yml:66]: In the future we could easily swap this to another site

File: packages/backend/package.json [hunk 4]
```diff
@@ -35,7 +35,7 @@
     "publish:local": "npx yalc push --replace --sig",
     "build:lib": "tsup --env.NODE_ENV production",
     "build:tests": "tsc -p tsconfig.test.json",
-    "build:runtime": "rsync -r --include '*/' --include '*.js' --include '*.mjs' --exclude='*' src/runtime dist",
+    "build:runtime": "cpy 'src/runtime/**/*.{mjs,js,cjs}' dist/runtime",
```
Comment (LekoArts) [inline packages/backend/package.json:38]: I was using https://github.com/nektos/act to test this workflow locally and `rsync` isn't installed in every linux distro. So I switched to a package that'll work everywhere (and is easier to grok)

File: playground/nextjs/package.json [hunk 5]
```diff
@@ -6,15 +6,15 @@
...
-    "yalc:add": "yalc add -- @clerk/shared @clerk/types @clerk/backend @clerk/clerk-react @clerk/clerk-sdk-node @clerk/nextjs"
+    "lint": "next lint"
   },
   "dependencies": {
     "@clerk/backend": "latest",
     "@clerk/clerk-react": "latest",
     "@clerk/clerk-sdk-node": "latest",
     "@clerk/nextjs": "latest",
     "@clerk/shared": "latest",
+    "@clerk/themes": "latest",
```
Comment (LekoArts) [inline playground/nextjs/package.json:17]: `@clerk/themes` is used but not actually installed

File: scripts/install-site-in-isolation.mjs [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+
+SITE_PATH: ${SITE_PATH}
+ROOT_PATH: ${ROOT_PATH}
+FULL_SITE_PATH: ${FULL_SITE_PATH}
+TMP_FOLDER: ${TMP_FOLDER}
+FULL_TMP_FOLDER: ${FULL_TMP_FOLDER}`);
+
+  // Installing secco
+  await core.group('Installing secco (if not already installed)', async () => {
+    await $`command -v secco || (command -v sudo && sudo npm install -g secco@latest) || npm install -g secco@latest`;
```
Comment (BRKalow) [inline scripts/install-site-in-isolation.mjs:31]: ❓ why the need for `sudo` here?

File: scripts/install-site-in-isolation.mjs
Comment (BRKalow) [inline scripts/install-site-in-isolation.mjs:1]: (passing thought) This feels like the beginning of a util that could be used locally to create playgrounds from our integration templates. 😄

File: .github/workflows/preview.yml [hunk 7]
```diff
@@ -0,0 +1,131 @@
...
+      - name: Setup
+        id: config
+        uses: ./.github/actions/init
+        with:
+          turbo-signature: ${{ secrets.TURBO_REMOTE_CACHE_SIGNATURE_KEY }}
+          turbo-team: ${{ vars.TURBO_TEAM }}
+          turbo-token: ${{ secrets.TURBO_TOKEN }}
+          registry-url: "https://registry.npmjs.org"
+
+      - name: Build release
```
Comment (BRKalow) [inline .github/workflows/preview.yml:58]:
```suggestion
      - name: Build packages
```

File: .github/workflows/preview.yml [hunk 8]
```diff
@@ -0,0 +1,131 @@
...
+        run: npm install --global vercel@latest
+
+      - name: Pull Vercel environment information
+        run: |
+          cd $FULL_TMP_FOLDER
+          vercel pull --yes --environment=preview --token=${{ secrets.VERCEL_CLERK_COOKIE_TOKEN }}
+
+      - name: Copy clerk-js/dist into public/clerk-js of test site
+        run: |
+          cp -r $GITHUB_WORKSPACE/packages/clerk-js/dist $FULL_TMP_FOLDER/public/clerk-js
```
Comment (BRKalow) [inline .github/workflows/preview.yml:78]: Good thinking 👍

**Review by BRKalow — APPROVED** [review]
👏 Looks great! No blocking comments. This will be a nice workflow improvement.

Should we add info on this to the contributing doc as well? Or do you want to stick to the internal notion page?

File: scripts/install-site-in-isolation.mjs (hunk 6 above)
Comment (LekoArts) [reply scripts/install-site-in-isolation.mjs:31]: If you run this locally (which it is designed for, you can fully run this locally) this is sometimes necessary in some environments. Ideally you already have it installed globally anyways then

File: scripts/install-site-in-isolation.mjs
Comment (LekoArts) [reply scripts/install-site-in-isolation.mjs:1]: Yeah, you can already use it locally start to finish 👍

**Comment by LekoArts** [comment]
> Should we add info on this to the contributing doc as well

I can do that in another PR, I saw some needed unrelated changes there

**Pull request merged**